src/Components/DrillPage.js

```diff
@@ -1,25 +1,24 @@
-import React, { Component } from 'react';
-import { ScrollView, StyleSheet, View, Text, Button, ImageBackground, TouchableOpacity } from 'react-native';
+import React from 'react';
+import { ScrollView, StyleSheet, View, Text, Image, ImageBackground, TouchableOpacity } from 'react-native';
 import { connect } from 'react-redux';
 
 class DrillPage extends React.Component {
-
-_toggleFavorite() {
-  const action = { type: "TOGGLE_FAVORITE", value: this.state.dr }
-  this.props.dispatch(action)
-}
-
-componentDidUpdate() {
-  console.log("componentDidUpdate : ")
-  console.log(this.props.favoritesDrill)
-}
+  constructor(props) {
+    super(props);
+    this.state = { drill: props.drill };
+  }
```

Comment:
I prefer the more modern variant of intitializing the state:

```
state = {
  drill: this.props.drill
}
```

Use this instead of the constructor